js/components/dataset/resource/availability.vue

```diff
@@ -1,28 +1,58 @@
 <template>
-    <span v-if="availability === 'AVAILABLE'" class="badge bg-green">✓</span>
-    <span v-if="availability === 'NOT_AVAILABLE'" class="badge bg-red">×</span>
-    <span v-if="availability === 'UNKNOWN'" class="badge bg-gray">?</span>
+    <div :title="_('Click to trigger a new check')" v-if="!checking" @click.stop="refresh()">
+        <span v-if="availability === 'AVAILABLE'" class="badge bg-green">✓</span>
+        <span v-if="availability === 'NOT_AVAILABLE'" class="badge bg-red">×</span>
+        <span v-if="availability === 'UNKNOWN'" class="badge bg-gray">?</span>
+    </div>
+    <div v-if="checking">
+        <span class="badge bg-blue">↻</span>
+    </div>
 </template>
 
 <script>
+import API from 'api';
+
 export default {
     name: 'resource-availability',
     props: {
         resource: {
             type: Object,
             required: true,
+        },
+        dataset: {
+            type: Object,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            checking: false,
+            availability: null,
         }
     },
-    computed: {
-        availability() {
-            switch (this.resource.extras && this.resource.extras['check:available']) {
+    created() {
+        this.availability = this.computeAvailability(this.resource.extras && this.resource.extras['check:available']);
+    },
+    methods: {
+        computeAvailability(checkAvailable) {
+            switch (checkAvailable) {
                 case true:
                     return 'AVAILABLE';
                 case false:
                     return 'NOT_AVAILABLE';
                 default:
                     return 'UNKNOWN';
             }
+        },
+        refresh() {
+            this.checking = true;
+            API.datasets.check_dataset_resource({dataset: this.dataset.id, rid: this.resource.id}, res => {
+                this.availability = this.computeAvailability(res.obj['check:available']);
+                this.checking = false;
+            }, err => {
+                console.log('Something went wrong with the check', err);
```

Comment:
Set the availability to unknown in that case?

Reply:
👍